### PR TITLE
Opaque pointer: Update createShaderSelection function API

### PR DESCRIPTION
**Pull request by mariusz-sikora-at-amd**
Extending function API to accept result base type as a last parameter.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7911652c97615c2c2d1a8378fb35c3ad8f45b988
<details><summary>CTS tests (Failed: 0/286534)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.6, Gfx8</summary><ul>

<li>Passed: 36222/72653 (49.9%)</li>
<li>Failed: 0/72653 (0.0%)</li>
<li>Not Supported: 36431/72653 (50.1%)</li><li>Warnings: 0/72653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 40420/72700 (55.6%)</li>
<li>Failed: 0/72700 (0.0%)</li>
<li>Not Supported: 32280/72700 (44.4%)</li><li>Warnings: 0/72700 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 37135/72707 (51.1%)</li>
<li>Failed: 0/72707 (0.0%)</li>
<li>Not Supported: 35572/72707 (48.9%)</li><li>Warnings: 0/72707 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 34834/68474 (50.9%)</li>
<li>Failed: 0/68474 (0.0%)</li>
<li>Not Supported: 33640/68474 (49.1%)</li><li>Warnings: 0/68474 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — COMMENTED** [review]
LGTM, but one minor suggestion inline

File: llpc/lower/llpcSpirvLowerRayTracing.cpp [hunk 1]
```diff
@@ -1107,8 +1112,12 @@ void SpirvLowerRayTracing::createShaderSelection(Function *func, BasicBlock *ent
     pSwitch->addCase(m_builder->getInt32(moduleIds[i]), shaderBlock);
     m_builder->SetInsertPoint(shaderBlock);
     auto funcName = std::string("_") + getShaderStageAbbreviation(stage) + "_" + moduleIdStr;
-    Type *retType = (inResult != nullptr) ? inResult->getType()->getPointerElementType() : m_builder->getVoidTy();
-    Value *result = m_builder->CreateNamedCall(funcName, retType, args, {Attribute::NoUnwind, Attribute::AlwaysInline});
+    // TODO: Remove this when LLPC will switch fully to opaque pointers.
+    if (inResult) {
+      assert(IS_OPAQUE_OR_POINTEE_TYPE_MATCHES(inResult->getType(), inResultTy));
+    }
```
[the remark this answers is not in the document]
Comment (mariusz-sikora-at-amd) [reply llpc/lower/llpcSpirvLowerRayTracing.cpp]: Thanks !

**Review by nhaehnle — APPROVED** [review]

**Comment by mariusz-sikora-at-amd** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit af9a06a5a369cf364b4a188b2fbba01f61022659
<details><summary>CTS tests (Failed: 0/286530)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.6, Gfx8</summary><ul>

<li>Passed: 36222/72652 (49.9%)</li>
<li>Failed: 0/72652 (0.0%)</li>
<li>Not Supported: 36430/72652 (50.1%)</li><li>Warnings: 0/72652 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 40419/72699 (55.6%)</li>
<li>Failed: 0/72699 (0.0%)</li>
<li>Not Supported: 32280/72699 (44.4%)</li><li>Warnings: 0/72699 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 37135/72706 (51.1%)</li>
<li>Failed: 0/72706 (0.0%)</li>
<li>Not Supported: 35571/72706 (48.9%)</li><li>Warnings: 0/72706 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 34834/68473 (50.9%)</li>
<li>Failed: 0/68473 (0.0%)</li>
<li>Not Supported: 33639/68473 (49.1%)</li><li>Warnings: 0/68473 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**